developer/src/kmcmplib/src/CompilerInterfaces.cpp

```diff
@@ -0,0 +1,368 @@
+#include <comperr.h>
+#include "kmcmplib.h"
+#include "filesystem.h"
+#include "CheckFilenameConsistency.h"
+#include "CheckNCapsConsistency.h"
+#include "DeprecationChecks.h"
+#include "versioning.h"
+#include "../../../../common/windows/cpp/include/ConvertUTF.h"
+#include "../../../../common/windows/cpp/include/keymanversion.h"
+
+#define SetError(err)       { if(AddCompileError(err) || (err & CERR_FATAL)) return FALSE; }
+
+bool CompileKeyboardHandle(FILE* fp_in, PFILE_KEYBOARD fk);
+
+EXTERN bool kmcmp_SetCompilerOptions(KMCMP_COMPILER_OPTIONS* options) {
+  //printf("°°-> changed to SetCompilerOptions() of kmcmplib \n");
+  if(!options || options->dwSize < sizeof(KMCMP_COMPILER_OPTIONS)) {
+    return FALSE;
+  }
+  kmcmp::FShouldAddCompilerVersion = options->ShouldAddCompilerVersion;
+  return TRUE;
+}
+
+#ifdef __EMSCRIPTEN__
+
+/*
+  WASM interface for compiler message callback
+*/
+EM_JS(int, wasm_msgproc, (int line, int msgcode, char* text, char* context), {
+  const proc = globalThis[context];
+  if(!proc || typeof proc != 'function') {
+    console.log(`[${line}: ${msgcode}: ${UTF8ToString(text)}]`);
+    return 0;
+  } else {
+    return proc(line, msgcode, UTF8ToString(text));
+  }
+});
+
+int wasm_CompilerMessageProc(int line, uint32_t dwMsgCode, char* szText, void* context) {
+  char* msgProc = static_cast<char*>(context);
+  return wasm_msgproc(line, dwMsgCode, szText, msgProc);
+}
+
+EXTERN bool kmcmp_Wasm_SetCompilerOptions(int ShouldAddCompilerVersion) {
+  KMCMP_COMPILER_OPTIONS options;
+  options.dwSize = sizeof(KMCMP_COMPILER_OPTIONS);
+  options.ShouldAddCompilerVersion = ShouldAddCompilerVersion;
+  return kmcmp_SetCompilerOptions(&options);
+}
+
+EXTERN bool kmcmp_Wasm_CompileKeyboardFile(char* pszInfile,
+  char* pszOutfile, int ASaveDebug, int ACompilerWarningsAsErrors,
+	int AWarnDeprecatedCode, char* msgProc
+) {
+  return kmcmp_CompileKeyboardFile(
+    pszInfile,
+    pszOutfile,
+    ASaveDebug,
+    ACompilerWarningsAsErrors,
+    AWarnDeprecatedCode,
+    wasm_CompilerMessageProc,
+    msgProc
+  );
+}
+#endif
+
+EXTERN bool kmcmp_CompileKeyboardFile(char* pszInfile,
+  char* pszOutfile, bool ASaveDebug, bool ACompilerWarningsAsErrors,
+	bool AWarnDeprecatedCode, kmcmp_CompilerMessageProc pMsgproc, void* AmsgprocContext
+) {
+  FILE* fp_in = NULL;
+  FILE* fp_out = NULL;
+  KMX_CHAR str[260];
+
+  //printf("°°-> changed to CompileKeyboardFile() of kmcmplib \n");
+
+  kmcmp::FSaveDebug = ASaveDebug;
+  kmcmp::FCompilerWarningsAsErrors = ACompilerWarningsAsErrors;   // I4865
+  AWarnDeprecatedCode_GLOBAL_LIB = AWarnDeprecatedCode;
+
+  kmcmp::CompileTarget = CKF_KEYMAN;
+
+  if (!pMsgproc || !pszInfile || !pszOutfile) SetError(CERR_BadCallParams);
+
+  PKMX_STR p;
+
+  if ((p = strrchr_slash(pszInfile)) != nullptr)
+  {
+    strncpy(kmcmp::CompileDir, pszInfile, (int)(p - pszInfile + 1));  // I3481
+    kmcmp::CompileDir[(int)(p - pszInfile + 1)] = 0;
+  }
+  else
+    kmcmp::CompileDir[0] = 0;
```

Comment:
I'm seeing a mix of coding styles for braces..
```
if // l.90
{
}
else
```

and 
```
if {  // e.g. l. 105
}
```

We prefer the latter, right?

Reply:
Yes, where I edit, I try to fixup. If I haven't edited a line, it remains the old style (l.90). I may have missed some spots...